### PR TITLE
Support `PauliSentence` in `qml.equal`

### PR DESCRIPTION
**Pull request by dwierichs**
**Context:**
Currently `qml.equal` can not handle `PauliSentence` instances, and they are a bit cumbersome to compare manually.

**Description of the Change:**
Support `PauliSentence` comparison in `qml.equal`.

**Benefits:**
Extended feature.

**Possible Drawbacks:**
N/A

**Related GitHub Issues:**
n/a

### CODE REVIEW

File: pennylane/ops/functions/equal.py [hunk 1]
```diff
@@ -356,6 +356,42 @@ def _equal_operators(
     return True
 
 
+@_equal_dispatch.register
```
Comment (albi3ro) [inline pennylane/ops/functions/equal.py:359]: Should we also add a comparison for `PauliWord`?

**Review by albi3ro — COMMENTED** [review]
Should we also add a comparison for `PaulWord`?

File: pennylane/ops/functions/equal.py [hunk 2]
```diff
@@ -356,6 +356,42 @@ def _equal_operators(
...
+def _equal_paulisentence(
+    op1: qml.pauli.PauliSentence,
+    op2: qml.pauli.PauliSentence,
+    check_interface=True,
+    check_trainability=True,
+    rtol=1e-5,
+    atol=1e-9,
+):
+    if set(op1) != set(op2):
+        return f"The Pauli words in the two PauliSentences differ:\n{set(op1)}\n{set(op2)}"
```
Comment (albi3ro) [inline pennylane/ops/functions/equal.py:369]: Might be more 
```suggestion
        if difference1 := set(op1).difference(set(op2)):
            return f"op1 has {difference1} not present in op2"
        difference2 := set(op2).difference(set(op1))
        return f"op2 has {difference2} not present in op1"
```
A little more work, but might be a little more clear as to the source of the difference when the objects get larger.

Comment (dwierichs) [reply pennylane/ops/functions/equal.py:369]: Added a version that always prints all non-vanishing differences. Thanks!

File: pennylane/ops/functions/equal.py [hunk 3]
```diff
@@ -356,6 +356,74 @@ def _equal_operators(
...
+):
+    if op1 != op2:
+        if set(op1) != set(op2):
+            err = "Different wires in Pauli words."
+            diff12 = set(op1).difference(set(op2))
+            diff21 = set(op2).difference(set(op1))
+            if diff12:
+                err += " op1 has {diff12} not present in op2."
+            if diff21:
+                err += " op2 has {diff21} not present in op1."
```
Comment (albi3ro) [inline pennylane/ops/functions/equal.py:374]:
```suggestion
                err += f" op1 has {diff12} not present in op2."
            if diff21:
                err += f" op2 has {diff21} not present in op1."
```

File: pennylane/ops/functions/equal.py [hunk 4]
```diff
@@ -356,6 +356,74 @@ def _equal_operators(
...
+    atol=1e-9,
+):
+    if set(op1) != set(op2):
+        err = "Different Pauli words in PauliSentences."
+        diff12 = set(op1).difference(set(op2))
+        diff21 = set(op2).difference(set(op1))
+        if diff12:
+            err += " op1 has {diff12} not present in op2."
+        if diff21:
+            err += " op2 has {diff21} not present in op1."
```
Comment (albi3ro) [inline pennylane/ops/functions/equal.py:400]:
```suggestion
            err += f" op1 has {diff12} not present in op2."
        if diff21:
            err += f" op2 has {diff21} not present in op1."
```

File: doc/releases/changelog-dev.md [hunk 5]
```diff
@@ -198,6 +198,9 @@ such as `shots`, `rng` and `prng_key`.
 
 <h3>Improvements 🛠</h3>
 
+* `qml.equal` now support `PauliSentence` instances.
```
Comment (dwierichs) [inline doc/releases/changelog-dev.md:201]:
```suggestion
* `qml.equal` now supports `PauliWord` and `PauliSentence` instances.
```

File: tests/ops/functions/test_equal.py [hunk 6]
```diff
@@ -1211,6 +1211,105 @@ def test_not_equal_operator_measurement(self, op1, op2):
...
+        ps1 = qml.simplify(ps1).pauli_rep
+        ps2 = qml.simplify(ps2).pauli_rep
+
+        assert qml.equal(ps1, ps2) is res
+        assert qml.equal(ps1 * 0.6, ps2 * 0.6) is res
+        assert qml.equal(ps2, ps1) is res
+
+    @pytest.mark.torch
+    def test_trainability_and_interface(self):
+        """Test that trainability and interface is compared correctly."""
```
Comment (dwierichs) [inline tests/ops/functions/test_equal.py:1260]:
```suggestion
        """Test that trainability and interface are compared correctly."""
```

**Review by astralcai — COMMENTED** [review]
Can you also test `qml.assert_equal` and confirm that the correct reason are returned?

**Review by astralcai — APPROVED** [review]
LGTM 🎸

**Review by albi3ro — APPROVED** [review]

**Pull request merged**